pyscriptjs/src/components/pyrepl.ts

```diff
@@ -51,6 +53,42 @@ export function make_PyRepl(interpreter: InterpreterClient, app: PyScriptApp) {
             logger.debug(`element ${this.id} successfully connected`);
         }
 
+        get src() {
+            return this.getAttribute('src');
+        }
+
+        set src(value) {
+            this.setAttribute('src', value);
+        }
+
+        attributeChangedCallback(name: string, oldVal: string, newVal: string) {
+            if (name === 'src' && newVal !== oldVal) {
+                void this.loadReplSrc();
+            }
+        }
+
+        /**
+         * Fetch url from src attribute of py-repl tags and
+         * preload the code from fetch response into the Corresponding py-repl tag,
+         * but please note that they will not be pre-run unless you click the runbotton.
+         */
+        async loadReplSrc() {
+            try {
+                const response = await robustFetch(this.src);
+                if (!response.ok) {
+                    return;
+                }
+                const parentElement = this.querySelector("div[class='cm-content']");
```

Comment:
I've missed this one!!! ... if this is a `parentElement` you should use [this.closest(selector)](https://developer.mozilla.org/en-US/docs/Web/API/Element/closest) instead.

If it's not a parent element then the variable name is misleading.

Relying on unscoped `querySelector` behavior looks a bit too sketchy to me, but since `closest` is the right tool for the job, I'd say let's go for it?

Reply:
@WebReflection I mean, it's the parent element of<divclass='cm content '>. 😂I'll modify the variable name as soon as possible🏃‍♂️